extensions/namespace-lookup/src/test/java/io/druid/server/namespace/URIExtractionNamespaceFunctionFactoryTest.java

```diff
@@ -300,6 +317,23 @@ public void simpleTest() throws IOException, ExecutionException, InterruptedExce
   }
 
   @Test
+  public void testReverseFunction() throws InterruptedException
+  {
+    Assert.assertNull(reverseFnCache.get(namespace.getNamespace()));
+    NamespaceExtractionCacheManagersTest.waitFor(manager.schedule(namespace));
+    Function<String, List<String>> reverseFn = reverseFnCache.get(namespace.getNamespace());
+    while (reverseFn == null) {
+      Thread.sleep(1);
```

Comment:
im really scared by sleep statements such as this. They cause random tests to fail in Travis. Can we use a latch to test instead?